modules/api/src/main/java/org/apache/ignite/table/criteria/CriteriaQuerySource.java

```diff
@@ -59,7 +60,8 @@ default Cursor<T> query(@Nullable Transaction tx, @Nullable Criteria criteria) {
      * @param tx Transaction to execute the query within or {@code null} to run within implicit transaction.
      * @param criteria The predicate to filter entries or {@code null} to return all entries from the underlying table.
      * @return Future that represents the pending completion of the operation.
-     * @throws IgniteException If failed.
+     * @throws CriteriaException If failed.
+     * @throws IllegalArgumentException If the names of the provided columns are incorrect.
```

Comment:
Async methods return futures. 
Imho, async method should never throw an exception. They should return failed future instead.

Reply:
Replaced with failed future